drivers/watchdog/Kconfig

```diff
@@ -13,6 +13,7 @@ if WATCHDOG
 
 config WDT_DISABLE_AT_BOOT
 	bool "Disable at boot"
+	depends on !SOC_COMPATIBLE_NRF
```

Comment:
I'm wondering if we should instead introduce an invisible Kconfig `HAS_WDT_DISABLE_AT_BOOT` similar to `HAS_WDT_MULTISTAGE` which can be selected if supported by that particular MCU or driver. That would be a more generic and scalable approach.

Reply:
Done. Please take a look.